### DISCUSSION TITLE
Field validation

### DISCUSSION THREAD

**Discussion by vafadarnaghi** [prisma/prisma#2852] [Q&A] [answered]
Hi all, 

First of all thank you for great tool like Prisma.

My question is about data validation/constraints in data modeling level. Would not it be useful if we could add validation rules to individual field in the model with the help of attributes, like it is already done for unique constraint?

For example, if column length set to 16, and query intends to write string with17 symbols, in this case Prisma client should not hit database and return corresponding error.

Also these validations / constraints could be projected at database level when creating database with prisma models from scratch (column length, uuid etc).

Example validation types could be: email, string length, integer length, greater than, less than etc.

Ofcouse, these are not new ideas, many ORMs already implements it. I think it would be great if Prisma also has its own solution.

Thank you.

**Comment by ryands17** [comment, marked as answer]
Hey @vafadarnaghi :wave: 
Currently Prisma doesn't support field level validations, so could you open a feature request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) regarding the above use case.

**Comment by ludralph** [comment]
Hi @vafadarnaghi,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
